lib/Backend/GroupBackend.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * @copyright Copyright (c) 2016, ownCloud, Inc.
+ *
+ * @author Christoph Wurst <christoph@winzerhof-wurst.at>
+ * @author Knut Ahlers <knut@ahlers.me>
+ * @author Roeland Jago Douma <roeland@famdouma.nl>
+ * @author Vincent Petry <vincent@nextcloud.com>
+ *
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program. If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+namespace OCA\VO_Federation\Backend;
+
+use OCP\Group\Backend\ABackend;
+use OCP\Group\Backend\IGetDisplayNameBackend;
+use OCP\Group\Backend\IGroupDetailsBackend;
+
+use OCP\ILogger;
+
+/**
+ * Abstract base class for user management
+ */
+class GroupBackend extends ABackend implements
+	IGetDisplayNameBackend,
+	IGroupDetailsBackend {
+
+	/**
+	 * @var string The application name.
+	 */
+	private $appName;
```

Comment:
```suggestion
	private string $appName;
```

Please strong type properties